filament/src/details/Texture.cpp

```diff
@@ -528,20 +516,53 @@ void FTexture::updateLodRange(uint8_t baseLevel, uint8_t levelCount) noexcept {
     }
 }
 
+void FTexture::setHandles(backend::Handle<backend::HwTexture> handle) noexcept {
+    assert_invariant(!(mHandle && !mHandleForSampling));
+    if (mHandle) {
+        mDriver->destroyTexture(mHandle);
+    }
+    if (mHandleForSampling != mHandle) {
+        mDriver->destroyTexture(mHandleForSampling);
+    }
+    mHandle = handle;
+    mHandleForSampling = handle;
+}
+
+void FTexture::setHandleForSampling(backend::Handle<backend::HwTexture> handle) const noexcept {
+    assert_invariant(!(mHandle && !mHandleForSampling));
+    if (mHandleForSampling && mHandleForSampling != mHandle) {
+        mDriver->destroyTexture(mHandleForSampling);
+    }
+    mHandleForSampling = handle;
+}
+
+backend::Handle<backend::HwTexture> FTexture::createPlaceholderTexture(
+        backend::DriverApi& driver) noexcept {
+    auto handle = driver.createTexture(
+            Sampler::SAMPLER_2D, 1, InternalFormat::RGBA8, 1, 1, 1, 1, Usage::DEFAULT);
+    static uint8_t pixels[4] = { 0, 0, 0, 0 };
+    driver.update3DImage(handle, 0, 0, 0, 0, 1, 1, 1,
+            { (char*)&pixels[0], sizeof(pixels),
+                    Texture::PixelBufferDescriptor::PixelDataFormat::RGBA,
+                    Texture::PixelBufferDescriptor::PixelDataType::UBYTE });
+    return handle;
+}
+
 backend::Handle<backend::HwTexture> FTexture::getHwHandleForSampling() const noexcept {
+    if (UTILS_UNLIKELY(mTarget == SamplerType::SAMPLER_EXTERNAL && !mHandleForSampling)) {
+        setHandleForSampling(createPlaceholderTexture(*mDriver));
```

Comment:
setHandleForSampling() could return `mHandleForSampling`, so here you could `return setHandleForSampling()`